### ISSUE TITLE
ipwidgets gui functions error

### ISSUE THREAD

**Issue by magnunor** [hyperspy/hyperspy#2227]
Using the most current pypi version of HyperSpy (1.5) and gui libraries (1.2), I get an error when running the following in a notebook:

```python
%matplotlib notebook
import hyperspy.api as hs
hs.preferences.gui()
```

Gives the error: https://gist.github.com/magnunor/ed2b75983b4735c485ffae78b09a26fd

This also affects things like `s.remove_background`.

This does not happen when using the traitsui gui (%matplotlib qt).

**Comment by ericpre** [comment]
It seems that this is a regression in calling the GUIs. A workaround is to set them explicitly by:
```python
hs.preferences.GUIs.enable_traitsui_gui = False
hs.preferences.GUIs.enable_ipywidgets_gui = True
hs.preferences.save()
```
before calling any GUIs.

Or to specify the toolkit to use by the GUI, for example:
```
s.remove_background(toolkit='ipywidgets')
hs.preferences.gui(toolkit='ipywidgets')
```

This happen only with the notebook, because the traitsui can not be used with the notebook matplotlib backend and it should not be called with this backend.

**Comment by magnunor** [comment]
That indeed fixed it! I'm guessing this also applies to the current anaconda release and the bundle release? (I can't currently test this on my own).

If yes, I'll have to include the workaround in the EELS notebook for the M&M short course.

**Comment by francisco-dlp** [comment]
Fixed in #2231

**Comment by ericpre** [comment]
#2231 is merged.
